workers-scaling/README.md

```diff
@@ -0,0 +1,88 @@
+and the workers have no more than one core.
+We definitely don't want to change the prefetch multiplier because that might
+cause [p-s#375](https://github.com/packit/packit-service/issues/375)
+for no significant benefit, but having more concurrent worker threads
+should hopefully be safe.
+
+Whether it's really safe depends on how Celery
+fetches messages for the worker's threads. If it waits for all threads
+to finish their tasks before fetching a new batch of messages (times prefetch-multiplier)
+for all threads then we might still hit [p-s#375](https://github.com/packit/packit-service/issues/375)
+if we have more workers serving the short-running-tasks queue.
+But if Celery fetches a new message(s) for a thread once it finishes a task
+without waiting for other threads then we should be fine (as long as prefetch-multiplier=1).
+I checked the [Celery source code](https://github.com/celery/celery)
+but couldn't find an answer for how/when it fetches messages for the threads
+(whether individually or for all at once). Someone's up for a challenge?;)
+But if we had only one worker (with more threads) serving the
+short-running-tasks queue then we should be fine anyway.
+
+#### Processes vs. threads aka. [Celery execution pool](https://www.distributedpython.com/2018/10/26/celery-execution-pool)
+
+- **Processes** (`prefork` pool) -
+  are good when tasks are CPU bound. The number of processes should not exceed
+  the number of CPUs on host.
+- **Threads** ([eventlet](https://docs.celeryq.dev/en/latest/userguide/concurrency/eventlet.html)/[gevent](https://www.gevent.org) pool) -
+  when tasks are I/O or network bound. The number of greenlets (green threads) is
+  unrelated to the number of CPUs on host and can go even to thousands.
+
+#### Autoscaling
+
+Regarding the `--concurrency` value, we even don't have to set a static number
+if we use [worker --autoscale](https://docs.celeryq.dev/en/latest/userguide/workers.html#autoscaling)
+which dynamically resizes the pool based on load.
+
+## Summary/Suggestion
+
+- Default for stg & dev can stay as it is, i.e. [one worker for all tasks](https://github.com/packit/deployment/blob/cb47f9e6f806f9e340669031c7eb58df02b164f1/playbooks/deploy.yml#L40)
+  with no concurrency/autoscaling/prefetching.
+- For prod, we should experiment with setting `--autoscale` (or just `--concurrency`)
+  together with `--pool=eventlet/gevent`. We should have only [one worker for serving
+  short-running tasks](https://github.com/packit/deployment/blob/cb47f9e6f806f9e340669031c7eb58df02b164f1/vars/packit/prod_template.yml#L77)
+  which would autoscale based on load and [no worker for all tasks](https://github.com/packit/deployment/blob/cb47f9e6f806f9e340669031c7eb58df02b164f1/vars/packit/prod_template.yml#L76)
+  to avoid [p-s#375](https://github.com/packit/packit-service/issues/375) regression.
```

Comment:
What about trying on stage first with multiple workers of the same type?

Reply:
Are you suggesting first trying your packit/deployment#361 on stage?
(your packit/deployment#361 would definitely work, but it is manual and takes much more resources (x pods vs. 1 pod with x threads))

Or to try on stage some variant of what I suggest here for prod?

Reply:
Sorry for late responses... I meant the second option -- try your changes on stage. (And add more workers to stage so the multi-worker processing is used.)